src/main/scala/com/advancedtelematic/director/http/AdminResource.scala

```diff
@@ -71,10 +69,16 @@ class AdminResource(extractNamespace: Directive1[Namespace], keyserverClient: Ke
       directorRepo.findOrCreate(namespace, keyType).map(StatusCodes.Created -> _)
     }
 
+  private val malformedRequestContentRejectionHandler = RejectionHandler.newBuilder().handle {
+    case MalformedRequestContentRejection(msg, _) => complete((StatusCodes.BadRequest, msg))
+  }.result()
```

Comment:
Is this not the default? What is the default without this handler?

Reply:
This specifically only handles the MalformedRequestContentRejection rejection, the default handler (https://github.com/advancedtelematic/libats/blob/667ffe7265270c3526c3d71caea810ce36be6f95/libats-http/src/main/scala/com/advancedtelematic/libats/http/DefaultRejectionHandler.scala#L28) does way more. Among other things it handles an `RequestEntityExpectedRejection`, which it shouldn't in my case.

Reply:
It can't fall back to the standard error handler, because that has the undesired side effect I was describing above. Optional entities seem to be way too much effort in Akka Http, optional query parameters are supported out of the box.

Reply:
Don't "errors inside the block using malformedRequestContentRejectionHandler" bubble up to the top-level error handlers?

Reply:
No, errors inside the block will use `malformedRequestContentRejectionHandler` which does not handle any other errors than MalformedRequestContentRejection, so it will just throw an exception and will be handled by the akka error handler.

Reply:
I meant rejections, sorry.

Reply:
You could add `.withFallback(RejectionHandler.default)` like in https://github.com/advancedtelematic/libats/blob/667ffe7265270c3526c3d71caea810ce36be6f95/libats-http/src/main/scala/com/advancedtelematic/libats/http/DefaultRejectionHandler.scala#L28 to fix it, but IMO messing with the error handling is not a very good idea

Reply:
You mean rejection handling? withFallback(RejectionHandler.default) doesn't work here, because it would handle RequestEntityExpectedRejection and complete with an error for it, which I don't want.

Reply:
I don't understand the point of this then. What is the problem we are trying to solve?

If the user sends an invalid key type, that should be a 400 anyway. If I send something like `keytype: bananas` I expect the server to either use bananas or fail, I don't expect it to use RSA because bananas are not available. If I don't send anything then ok, i expect it to use whatever the default is, in this case RSA.

Reply:
That's exactly what my rejection handler is for. Without it the default key would be used for either sending nothing or a wrong key type. So it would use RSA for the key type "ed25519".

Reply:
Ah! Ok. I was not getting it, sorry.

Then ok, only thing I'd change is 
`case MalformedRequestContentRejection(msg, _) => complete((StatusCodes.BadRequest, ErrorRepresentation(ErrorCodes.InvalidEntity, msg))))` to be consistent with the other `handler` blocks on the default handler.

Reply:
Maybe it would be clearer to use a Directive like this one: https://github.com/akka/akka-http/issues/284#issuecomment-246327417

Reply:
The thing is the moment you use `entity` it would reject the route and the only way to edit the rejection list after that is to use a rejection handler. We could use a Directive that does what `entity` does but does not reject and completes with error instead, but it's more complicated than your solution.

Reply:
The person who wrote that directive makes it sound like it would work, but I haven't tried it.

Reply:
But how would it work? It would work only when the entity was not provided, but not with an invalid entity I think?